src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -330,8 +341,40 @@ def _get_pool_status(
                 });
 
         return (ret, root_map, pool_root)
-
-
+    
+    def _update_progress_events(self):
+        osdmap = self.get_osdmap()
+        pools = osdmap.get_pools()
+        for pool_id in list(self._event):
+            ev = self._event[pool_id]
+            pool_data = pools[int(pool_id)]
+            pg_num = pool_data['pg_num']
+            pg_num_target = pool_data['pg_num_target']
+            initial_pg_num = ev._pg_num
+            initial_pg_num_target = ev._pg_num_target
+            if pg_num == pg_num_target:
+                self.remote('progress', 'complete', ev._ev_id)
+                del self._event[pool_id]
+                continue
+            if pg_num == initial_pg_num:
+                continue
+            if pg_num < pg_num_target:
+                self.remote('progress', 'update', ev._ev_id, 
+                                    ev_msg=" PG autoscaler increasing pool %s PGs from %d to %d" % 
+                                    (pool_id, pg_num, pg_num_target), 
+                                    ev_progress=pg_num/pg_num_target,
```

Comment:
I'm thinking the progress should be measured by current pg_num compared to distance between the last initial_pg_num when the module made a change, and target_pg_num

i.e. for increasing, ```(pg_num - initial_pg_num)/(target_pg_num - initial_pg_num)```

Each time the autoscaler chose a new target, it could reset the initial_pg_num and target_pg_num for an existing event. Then the progress is measuring number of pgs changed in the latest decision, which makes sense since earlier history of shrinking/growing pg_num becomes irrelevant when the autoscaler changes things again.

Then in this function, the PgAdjustmentProgress wouldn't change, only the remote ev_progress would be updated.